client/src/modules/dashboard/Chapters/pages/ChaptersPage.tsx

```diff
@@ -50,28 +90,55 @@ export const ChaptersPage: NextPageWithLayout = () => {
   const isLoading = loading || loadingUser || !data;
   if (isLoading || error) return <DashboardLoading error={error} />;
 
+  // Works
+  // const filteredChapter =  data?.dashboardChapters.filter(({ name }) =>
+  //     name.toLowerCase().includes(search),
+  //   );
+
+  // Works, if I change the call to filteredChapter()
+  // const filteredChapter = () =>  data?.dashboardChapters.filter(({ name }) =>
+  // name.toLowerCase().includes(search)
+  // );
+
+  // Fails, but why?
+  const filteredChapter = useMemo(
+    () =>
+      data.dashboardChapters.filter(({ name }) =>
+        name.toLowerCase().includes(search),
+      ),
+    [data, search],
+  );
```

Comment:
Is this failing because it relies on a condition? The hooks run on mount, but the data is got on load?

Reply:
RIP performance, I guess 🤣. maybe I can move the logic into the top and create a function that run it, but idk if that actually worth it, while this works